lib/block-supports/layout.php

```diff
@@ -306,14 +306,22 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 			 * They're added separately because padding might only be set on one side.
 			 */
 			if ( isset( $block_spacing_values['declarations']['padding-right'] ) ) {
-				$padding_right   = $block_spacing_values['declarations']['padding-right'];
+				$padding_right = $block_spacing_values['declarations']['padding-right'];
+				// Add unit if 0.
+				if ( '0' === $padding_right ) {
```

Comment:
Are we guaranteed to get a string back from `gutenberg_style_engine_get_styles`?

Reply:
At a quick glance, the value is passed through the style engine's [`is_valid_style_value`](https://github.com/WordPress/gutenberg/blob/trunk/packages/style-engine/class-wp-style-engine.php#L354) function. Which should exclude an integer `0`.

```
		protected static function is_valid_style_value( $style_value ) {
			return '0' === $style_value || ! empty( $style_value );
		}
```

Reply:
Good question. I don't _think_ `gutenberg_style_engine_get_styles` will coerce the value from a quick look at the code, but padding values set by the UI will always be strings, and it's expected that values for spacing will be strings, I believe.

That said, this could be hardened in a follow-up to be `if ( '0' === $padding_right || 0 === $padding_right ) {` and similarly for the left padding.

Reply:
Thanks, I missed that one Aaron! 👍